server/helpers.js

```diff
@@ -22,10 +22,140 @@ async function getUserData(userId) {
   const trackFeatures = await prisma.spotifyMusicTrack.findMany();
 
   const { normalizedUserExpression, normalizedUserRecommendation, normalizedTrackFeatures } = normalizeData({ userExpression, userRecommendation, trackFeatures });
-  console.log(normalizedUserRecommendation);
+  console.log(normalizedUserRecommendation, normalizedUserExpression);
+
+  const recommendedTracks = recommendTracks({ 
+    normalizedUserExpression, 
+    normalizedUserRecommendation, 
+    normalizedTrackFeatures 
+  });
+  
+  console.log('Recommended tracks:', recommendedTracks[1]);
+
   return { normalizedUserExpression, normalizedUserRecommendation, normalizedTrackFeatures };
 }
-getUserData(1)
+
+
+function recommendTracks({ normalizedUserExpression, normalizedUserRecommendation, normalizedTrackFeatures }) {
```

Comment:
This function is pretty big. I think you can break it up into smaller functions that does smaller tasks and call it here. For example, you could break this function into several different functions like: 1 getting the criterias, 2 getting temp range, 3 combining criterias, 4 filtering

Reply:
Ok, That make sense, thanks for the feedback on this Lauren